src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1536,16 +1624,6 @@ bool OutputStateMachineEngine::_GetOscSetColorTable(const std::wstring_view stri
 
     bool foundTableIndex = false;
     bool success = false;
-    // We can have anywhere between [11,16] characters
-    // 11 "#;rgb:h/h/h"
-    // 16 "###;rgb:hh/hh/hh"
-    // Any fewer cannot be valid, and any more will be too many.
-    // Return early in this case.
-    //      We'll still have to bounds check when parsing the hh/hh/hh values
-    if (string.size() < 11 || string.size() > 16)

```

Comment:
With the introduction of XOrg color names support, this check does not offer meaningful shortcut. I think it's OK to remove it.

Reply:
You need at least a minimum size check, or adjustments to the loop below, otherwise it can throw an exception (which will cause the app to crash) when reading past the end of the string. Test case: `printf "\e]4;\e\\"`

Reply:
Thanks. I changed the loop and added some test cases to cover this.